### PR TITLE
chore: Amplify upgrade

### PR DESCRIPTION
**Pull request by mnischay**
# Description

updating amplify to v6 for mitigating accessibility issues and being up-to date with amplify

Fixes # (issue)

1. updated amplify ui and core version to v6
2. updated auth flow tests

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] tested the deployed instance for login
- [x] all the CRUD operations work
- [x] logout works
- [x] Migration apis are not blowing up
- [x] Reload on the dashboard page is working

https://github.com/awslabs/iot-application/assets/135036199/98f2fa38-313d-40fd-ba8b-38dd5ea32833

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: tests/auth.setup.ts [hunk 1]
```diff
@@ -19,6 +20,16 @@ setup('authenticate', async ({ page }) => {
   // user clicks sign-in
   await page.getByRole('button', { name: 'Sign in' }).click();
 
+  const [index] = await waitForOneOf([
+    page.getByText('Skip', { exact: true }),
```
Comment (chejimmy) [inline tests/auth.setup.ts:24]: what is this 'Skip' text?

File: apps/client/src/auth/auth-service.interface.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export interface AuthService {
    * Returns the user's session token.
    * @return - the user's session token
    */
-  getToken(): Promise<string>;
+  getToken(): Promise<string | undefined>;
```
[the remark this answers is not in the document]
Comment (mnischay) [reply apps/client/src/auth/auth-service.interface.ts:22]: yeah, debated about this. we do throw error so the error handler should catch it and render the error page...

**Review by chejimmy — APPROVED** [review]
LGTM TY 🙏

File: apps/client/src/auth/auth-service.interface.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mnischay) [reply apps/client/src/auth/auth-service.interface.ts]: if we throw error , we will not even return undefined, so you are right , it cannot be undefined, i was debating to throw error or handle it differently and the undefined was left over from the previous approach, so good to remove it.

**Review by chejimmy — APPROVED** [review]

**Pull request merged**